lib/file.c

```diff
@@ -196,8 +198,33 @@ static CURLcode file_connect(struct Curl_easy *data, bool *done)
     return CURLE_URL_MALFORMAT;
   }
 
+  #if defined(AMIGA_FILESYSTEM)
+  /*
+   * A leading slash in an AmigaDOS path denotes the parent
+   * directory, and hence we block this as it is relative.
+   * Absolute paths start with 'volumename:', so we check for
+   * this first. Failing that, we treat the path as a real unix
```

Comment:
Can the volume name be any length? Or maybe it doesn't matter as the open will just fail anyway if tried with a too long name?

Reply:
> Can the volume name be any length? Or maybe it doesn't matter as the open will just fail anyway if tried with a too long name?

I'm not sure on the length limit (probably 107 characters, at a guess), but it may also differ depending which filesystem is used. Normally, volume names are not that long though, but equally always longer than the drive letter than Windows uses.